rct229/rulesets/ashrae9012019/section5/section5rule38.py

```diff
@@ -0,0 +1,251 @@
+
+
+class Section5Rule38(RuleDefinitionListIndexedBase):
+    """Rule 38 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule38, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section5Rule38.BuildingRule(),
+            index_rmr="baseline",
+            id="5-38",
+            description="Skylight SHGC properties shall match the appropriate requirements in Tables G3.4-1 through G3.4-8 using the value and the applicable skylight percentage.",
+            ruleset_section_title="Envelope",
+            standard_section="Section G3.1-5(e) Building Envelope Modeling Requirements for the Baseline building",
+            is_primary_rule=True,
+            list_path="ruleset_model_instances[0].buildings[*]",
+            data_items={"climate_zone": ("baseline", "weather/climate_zone")},
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+        )
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule38.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                list_path="$..surfaces[*]",
+                each_rule=Section5Rule38.BuildingRule.RoofRule(),
+                index_rmr="baseline",
+                manual_check_required_msg=MANUAL_CHECK_MSG,
+            )
+
+        def manual_check_required(self, context, calc_vals=None, data=None):
+            building_b = context.baseline
+            climate_zone = data["climate_zone"]
+
+            building_scc_skylight_roof_ratios_dict_b = (
+                get_building_scc_skylight_roof_ratios_dict(climate_zone, building_b)
+            )
+
+            target_exterior_2per_residential = table_G34_lookup(
+                climate_zone,
+                "EXTERIOR RESIDENTIAL",
+                "SKYLIGHT",
+                skylit_wwr=0.0,
+            )["solar_heat_gain_coefficient"]
+
+            target_exterior_above2_residential = table_G34_lookup(
+                climate_zone,
+                "EXTERIOR RESIDENTIAL",
+                "SKYLIGHT",
+                skylit_wwr=0.021,
+            )["solar_heat_gain_coefficient"]
+
+            target_exterior_2per_nonresidential = table_G34_lookup(
+                climate_zone,
+                "EXTERIOR NON-RESIDENTIAL",
+                "SKYLIGHT",
+                skylit_wwr=0.021,
+            )["solar_heat_gain_coefficient"]
+
+            target_exterior_above2_nonresidential = table_G34_lookup(
+                climate_zone,
+                "EXTERIOR NON-RESIDENTIAL",
+                "SKYLIGHT",
+                skylit_wwr=0.021,
+            )["solar_heat_gain_coefficient"]
+
+            return building_scc_skylight_roof_ratios_dict_b[
+                SCC.EXTERIOR_MIXED
+            ] > 0 and (
```

Comment:
This can also rewrite into:

```python
all([building_scc_skylight_roof_ratios_dict_b[SCC.EXTERIOR_MIXED] > 0,
  target_exterior_2per_residential==target_exterior_above2_residential,
  target_exterior_2per_nonresidential == target_exterior_above2_nonresidential,
  target_exterior_2per_residential == target_exterior_2per_nonresidential
])
```

Reply:
Great point! Addressed.